crates/services/p2p/src/request_response/messages.rs

```diff
@@ -28,27 +34,43 @@ pub enum RequestMessage {
     Transactions(Range<u32>),
 }
 
-/// Holds oneshot channels for specific responses
-#[derive(Debug)]
-pub enum ResponseChannelItem {
-    Block(oneshot::Sender<Option<SealedBlock>>),
-    SealedHeaders(oneshot::Sender<(PeerId, Option<Vec<SealedBlockHeader>>)>),
-    Transactions(oneshot::Sender<Option<Vec<Transactions>>>),
-}
-
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub enum ResponseMessage {
-    Block(Option<SealedBlock>),
+    Block(Option<Arc<SealedBlock>>),
     SealedHeaders(Option<Vec<SealedBlockHeader>>),
     Transactions(Option<Vec<Transactions>>),
 }
 
+pub type OnResponse<T> = oneshot::Sender<Result<T, ResponseError>>;
+
+#[derive(Debug)]
+pub enum TypedResponseChannel {
```

Comment:
These aren't channels. The old name `ResponseChannelItem` isn't the best. "Item" isn't a helpful word, but it's the Item that comes over the "response channel", so it's accurate.

`ChannelResponse`? I'd prefer something that describes what the channel is used for, rather than just saying it is used by a channel.

Reply:
Renaemd to `ResponseSender`